File: src/main/java/net/lumalyte/statistics/StatType.java
package net.lumalyte.statistics;

/**
 * Enumeration of different statistic types that can be tracked and used for leaderboards.
 * 
 * @author LumaLyte
 * @version 1.0
 * @since 1.0
 */
public enum StatType {
    
    /** Total number of games won */
    WINS,
    
    /** Total number of kills */
    KILLS,
    
    /** Total number of games played */
    GAMES_PLAYED,
    
    /** Kill/Death ratio (calculated) */
    KILL_DEATH_RATIO,
    
    /** Win rate percentage (calculated) */
    WIN_RATE,
    
    /** Total time played in seconds */
    TIME_PLAYED,
    
    /** Best placement achieved (lower is better) */
    BEST_PLACEMENT,
    
    /** Best win streak achieved */
    WIN_STREAK,
    
    /** Number of top 3 finishes */
    TOP3_FINISHES,
    
    /** Total damage dealt to other players */
    DAMAGE_DEALT,
    
    /** Number of chests opened */
    CHESTS_OPENED
} 

File: src/main/kotlin/net/lumalyte/lumasg/domain/StatType.kt
package net.lumalyte.lumasg.domain

/** Types of statistics that can be used for leaderboard filtering. */
enum class StatType(val displayName: String, val columnName: String) {
    WINS("Wins", "wins"),
    KILLS("Kills", "kills"),
    GAMES_PLAYED("Games Played", "games_played"),
    KILL_DEATH_RATIO("K/D Ratio", "kills"), // computed from kills/deaths
    WIN_RATE("Win Rate", "wins"), // computed from wins/gamesPlayed
    TIME_PLAYED("Time Played", "total_time_played"),
    BEST_PLACEMENT("Best Placement", "best_placement"),
    WIN_STREAK("Win Streak", "best_win_streak"),
    TOP3_FINISHES("Top 3 Finishes", "top3_finishes"),
    DAMAGE_DEALT("Damage Dealt", "damage_dealt"),
    CHESTS_OPENED("Chests Opened", "chests_opened");
}
